### PR TITLE
fix: fix enablePartialUpdates fields check, closes #128

### LINKED ISSUE #128: enablePartialUpdates cannot find the match field
On build, I'm getting this error.

```
when enablePartialUpdates is true, the objects must have at least one of the match fields. Current object:
{
  "objectID": "e2a187c2-1450-5ba3-bdc1-f5572ae8c117",
  "id": "e2a187c2-1450-5ba3-bdc1-f5572ae8c117",
  "title": "Peppermint Pattie Cupcakes",
  "slug": "peppermint-pattie-cupcakes",
  "publishDate": "February 14, 2021",
  "tags": [
    {
      "title": " Vegetarian"
    },
  ],
  "heroImage": {
    "title": "peppermint-pattie-cupcakes-1",
    "gatsbyImageData": {
      "images": {
        "sources": [
          {
            "srcSet": "//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=156&h=234&q=100&fm=webp
156w,\n//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=313&h=469&q=100&fm=webp
313w,\n//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=625&h=937&q=100&fm=webp 625w",
            "sizes": "(min-width: 625px) 625px, 100vw",
            "type": "image/webp"
          }
        ],
        "fallback": {
          "src": "//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=625&h=937&q=100&fm=png",
          "srcSet": "//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=156&h=234&q=100&fm=png
156w,\n//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=313&h=469&q=100&fm=png
313w,\n//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=625&h=937&q=100&fm=png 625w",
          "sizes": "(min-width: 625px) 625px, 100vw"
        }
      },
      "layout": "constrained",
      "width": 900,
      "height": 1349
    },
    "ogimg": {
      "src": "//images.ctfassets.net/frks0qvtc4r5/4pszADie5eB6G2Kk5dFfLT/d16fe00a7b5bf86119d4f02621d4c2a9/peppermint-pattie-cupcakes-1.png?w=900&q=50"
    }
  },
  "body": {
    "childMarkdownRemark": {
      "html": "<p>......</p>",
      "excerpt": "...…"
    }
  },
  "recipe": {
    "raw": "{\"data\":{},\"content\":
        ......
        ......
        ......
        ......
    with a quarter of a York candy, and enjoy!\",\"nodeType\":\"text\"}],\"nodeType\":\"paragraph\"}],\"nodeType\":\"list-item\"}],\"nodeType\":\"ordered-list\"},{\"data\":{},\"content\":[{\"data\":{},\"marks\":[],\
    "value\":\"\",\"nodeType\":\"text\"}],\"nodeType\":\"paragraph\"}],\"nodeType\":\"document\"}"
      }
}
expected one of these fields:
slug
modified
```

And in my gatsby-config file:
```
{

      resolve: `gatsby-plugin-algolia`,
      options: {
        appId: process.env.ALGOLIA_APP_ID,
        apiKey: process.env.ALGOLIA_API_KEY,
        indexName: process.env.ALGOLIA_INDEX_NAME, // for all queries
        queries,
        chunkSize: 10000, // default: 1000
        enablePartialUpdates: true, // default: false
        matchFields: ['slug', 'modified'], // Array<String> default: ['modified']
        concurrentQueries: true, // default: true
        skipIndexing: false, // default: false, useful for e.g. preview deploys or local development
      },
```

The plugin can't seem to identify the slug field. Any help would be much appreciated!

### PR DESCRIPTION
**Pull request by thecodingwizard**
https://github.com/algolia/gatsby-plugin-algolia/blob/master/gatsby-node.js#L169 looks like it should be `.some` instead of `.every`.

closes #128

### CODE REVIEW

**Comment by Haroenv** [comment]
I'm not sure this is right, the error is thrown when `(!every)`, did you want to change it to `(some)` instead of `(!some)`?

**Comment by thecodingwizard** [comment]
We want to fail if no property in match fields exists in the object, right? So if .some returns false then no field exists ==> fail, if .some returns true then at least one field exists ==> pass.

Or is the intended behavior that all fields need to exist (in which case .every is correct, but the error message should be updated?)

(oops I haven't tested this code though)

File: gatsby-node.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ async function runIndexQueries(
         if (queryResultsMap.hasOwnProperty(id)) {
           // key matches fresh objects, so compare match fields
           const newObj = queryResultsMap[id];
-          if (!matchFields.every(field => newObj.hasOwnProperty(field))) {
+          if (!matchFields.some(field => newObj.hasOwnProperty(field))) {
```
Comment (Haroenv) [inline gatsby-node.js:169]:
```suggestion
          if (matchFields.every(field => newObj.hasOwnProperty(field) === false)) {
```

(that's the same as your fix, but it makes the code easier to read IMO)

**Comment by Haroenv** [comment]
You're right, I misread the condition there

File: gatsby-node.js (hunk 1 above)
Comment (thecodingwizard) [reply gatsby-node.js:169]: do you mean `.some` instead of `.every`?

Comment (Haroenv) [reply gatsby-node.js]: If every matchField is missing, we expect the condition, right? (Same as !some having the condition, opposite of the existing condition). It'll check this again tomorrow with a fresh mind

Comment (thecodingwizard) [reply gatsby-node.js]: oh, you're right. I misread and thought the comparator was outside the `every` function 🤦 I'll update the code :)

**Review by Haroenv — APPROVED** [review]

**Pull request merged**